python/GafferUI/Menu.py

```diff
@@ -405,11 +408,31 @@ def __evaluateItemValue( self, itemValue ) :
 
 		return itemValue
 
+	@GafferUI.LazyMethod()
+	def _lazyRepolish(self):
+
+		self._repolish()
+
+	@GafferUI.LazyMethod()
+	def __lazyUpdateSearchMenu( self, text):
+
+		return self.__updateSearchMenu( str(text) )
+
+	def __doUpdateSearchMenu( self, text ) :
+
+		# do a lazyUpdate in case we delete characters
+		if self.__previousSearchText[:-1] == str(text):
```

Comment:
I added the lazy update here because otherwise we have a bit of a delay when deleting the whole searchstring holding down backspace.
Seems like that delayed update is responsible though for the the issues you were running into (that deleting and adding back characters changes results and the errorstate)